### PR TITLE
Changes to get new features work for impersonate app

### PR DESCRIPTION
**Pull request by sharidas**
Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description
Changes to work with new changes proposed for impersonate app.

## How Has This Been Tested?

The changes has been tested by creating new include groups for impersonate app.
Logout functionality is also tested. Logout makes user reach back to the original user.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sharidas** [comment]
@PVince81 Kindly review my change.

File: core/templates/layout.user.php [hunk 1]
```diff
@@ -88,7 +88,13 @@
 					</li>
 				<?php endforeach; ?>
 					<li>
-						<a id="logout" <?php print_unescaped(OC_User::getLogoutAttribute()); ?>>
+						<a id="logout" <?php
+						if(\OC::$server->getAppConfig()->getValue('impersonate', 'enabled', '') !== '') {
```
Comment (PVince81) [inline core/templates/layout.user.php]: :-1: please, no app-specific code in core

File: lib/private/TemplateLayout.php [hunk 2]
```diff
@@ -63,6 +63,8 @@ public function __construct( $renderAs, $appId = '' ) {
 		// yes - should be injected ....
 		$this->config = \OC::$server->getConfig();
 
+		\OC::$server->getEventDispatcher()->dispatch('OC\TemplateLayout::loadAdditionalScripts');
```
Comment (PVince81) [inline lib/private/TemplateLayout.php]: this is fine, will likely be useful

File: core/templates/layout.user.php (hunk 1 above)
Comment (sharidas) [inline core/templates/layout.user.php]: sure. Will take care of this.

**Comment by PVince81** [comment]
@sharidas obsolete ?

**Comment by sharidas** [comment]
@PVince81 I was thinking about this change. Can I close this because we no longer need it.

**Pull request closed without merging**